### PR TITLE
Draft feat: Take less screenshots

### PR DESCRIPTION
**Pull request by ptbrowne**
Argos limits us to 500 screenshots so instead of screenshotting each
example, we screenshot each component page

One additional challenge introduced by screenshotting at component level, is that for some examples, we *need* to screenshot at example levels, for example with modals or action menus. To support this usecase, a configuration attribute `perExampleScreenshot` is introduced. When this is configured, screenshot are taken at example level.

### CODE REVIEW

File: scripts/screenshots.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const defaultGetScreenshotName = ({ component, viewport }) =>
 
 const formatViewport = viewport => `${viewport.width}x${viewport.height}`
 
+const pushAll = (arr, items) => arr.push.apply(arr, items)
```
Comment (y-lohse) [inline scripts/screenshots.js:32]: Nit, but here's a little trick: `array.push` accepts an infinite number of args, so `arr.push(...items)` does the same :)

Comment (ptbrowne) [reply scripts/screenshots.js:32]: Yes but it also makes a copy of the array and also needs a tad bit more code ;)

image

**Comment by y-lohse** [comment]
Mmh I don't think this works as intended, it seems like we don't have screenshots at an example level for ActionMenu and Modal. Are you seeing the same thing?

**Comment by ptbrowne** [comment]
Yes, after renaming the config attribute in code, I did not rename it in the config file, should be better now.

**Comment by y-lohse** [comment]
Nice, I believe this time we're good!

(The desktop action menu screenshots are too small but it's unrelated to this PR.)

**Pull request merged**